### PR TITLE
Add libjxl

### PR DESCRIPTION
**Pull request by sshockwave**
An alternative to #17812

Caveats:
- `lcms2` is not added since it is using jpeg-9. `libjxl` uses `jpeg_mem_*()` apis that are not compatible with jpeg-9.
  - https://github.com/libjxl/libjxl/blob/ec30ce72cdb7e4d05e204ff20ed824b00bc9fda1/doc/color_management.md#cms-library
  - https://github.com/libjxl/libjxl/issues/1734
- `skcms` is a single-file library that originates from [skia](https://skia.googlesource.com/skcms/). It is bundled using `git_url` since it's currently not independently distributed.
  - https://github.com/libjxl/libjxl/pull/441
- `cjxl` and `djxl` are separated into a different package `libjxl-tools` since they bring in a lot of other image codecs that are not required by the core `libjxl` library.

Build matrix:
- Platforms
  - [x] linux64
  - [x] win64
  - [x] osx64
- Codecs
  - [x] gif
  - [x] jpeg
  - [x] openexr
  - [x] png
  - [x] <del>webp & avif (only used in benchmarks)</del>

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sshockwave** [comment]
@conda-forge-admin please rerender

**Comment by sshockwave** [comment]
Missing runtime on Windows:
```
WARNING (libjxl-tools,Library/bin/djxl.exe): Needed DSO Library/bin/VCRUNTIME140.dll found in ['vs2015_runtime']
WARNING (libjxl-tools,Library/bin/djxl.exe): .. but ['vs2015_runtime'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (libjxl-tools,Library/bin/djxl.exe): Needed DSO Library/bin/api-ms-win-crt-stdio-l1-1-0.dll found in ['ucrt']
WARNING (libjxl-tools,Library/bin/djxl.exe): .. but ['ucrt'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
```
I built the package using clang instead of vs2019. Is this a concern?
@conda-forge-admin, please ping conda-forge/help-c-cpp

**Comment by sshockwave** [comment]
Alternative to #17812

@hmaarrfk I would like to hear your thoughts! Would you mind being on the maintainers' list?

File: recipes/libjxl/meta.yaml
Comment (hmaarrfk) [inline recipes/libjxl/meta.yaml:106]:
```suggestion
        - {{ pin_subpackage('libjxl', exact=True) }}
```

When you pin to subpackages, it is best to pin to exact, instead of loose things to ensure that the view is exactly consistent

File: recipes/libjxl/meta.yaml
Comment (hmaarrfk) [inline recipes/libjxl/meta.yaml:112]:
```suggestion
        - djxl --version
        - cjpeg_hdr --version
        - jxlinfo --version
```

we might as well add tests for these.

File: recipes/libjxl/meta.yaml
Comment (hmaarrfk) [inline recipes/libjxl/meta.yaml:121]: is it possible to build out this dependency as an other package or to disable this feature?

Comment (hmaarrfk) [reply recipes/libjxl/meta.yaml:122]: Thank you for the detailed explanation

File: recipes/libjxl/meta.yaml
Comment (hmaarrfk) [inline recipes/libjxl/meta.yaml]: Isnt this a useful executable?

Comment (sshockwave) [reply recipes/libjxl/meta.yaml]: Yes, it is. But it does not have a `--version` option. I'm making another test

File: recipes/libjxl/meta.yaml
Comment (sshockwave) [reply recipes/libjxl/meta.yaml]: I see. It looks like a good practice. Thanks!

File: recipes/libjxl/meta.yaml
Comment (hmaarrfk) [inline recipes/libjxl/meta.yaml:15]:
```suggestion
  # No tags, versions, or stable ABIs are exposed in the
  # [skcms repo](https://skia.googlesource.com/skcms/)
  # so I don't think it fits well in conda's versioning system.
  - git_url: https://skia.googlesource.com/skcms
```

File: recipes/libjxl/meta.yaml
Comment (hmaarrfk) [inline recipes/libjxl/meta.yaml:13]: we could probably add one https://github.com/conda-forge/openexr-feedstock/issues/27

Comment (sshockwave) [reply recipes/libjxl/meta.yaml:13]: I can try to push a fix

Comment (hmaarrfk) [reply recipes/libjxl/meta.yaml:13]: Ok, it should be available in a few hours.

**Review by hmaarrfk — APPROVED** [review]

**Comment by sshockwave** [comment]
We need another update https://github.com/conda-forge/imath-feedstock/pull/5

**Comment by hmaarrfk** [comment]
it seems to have worked already though.

**Comment by sshockwave** [comment]
No, it's building without OpenEXR:

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=615488&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=2fb5b0a7-737c-5f5c-8f3f-f6db6174bacf&l=1567

**Comment by hmaarrfk** [comment]
thank you for being thorough!

**Comment by hmaarrfk** [comment]
it may take time for the package to propagate. I'll be curious to see this working on windows. pkgconfig is not really standard there.

**Comment by sshockwave** [comment]
Oops. I was too eager to see the results and forgot that it takes time to build.

**Comment by hmaarrfk** [comment]
> Oops. I was too eager to see the results and forgot that it takes time to build.

It mostly takes time for the CDNs to get the package. That takes 1-2 hours.

**Comment by sshockwave** [comment]
LGTM

**Comment by sshockwave** [comment]
May I ask what's blocking the merge? I'd like to get a review @conda-forge/help-c-cpp

**Comment by hmaarrfk** [comment]
I guess I wanted to sleep in it

**Pull request merged**
